server/tests/controllers/articles.js

```diff
@@ -0,0 +1,182 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import server from '../../..';
+
+
+chai.should();
+
+chai.use(chaiHttp);
+
+// user expected to have been created by tests.controllers.users
+const user = {
+  username: 'testArticle',
+  firstName: 'Test',
+  lastName: 'Article',
+  email: 'test1@gmail.com',
+  password: '0000'
+};
+
+// tem test article
+const Article = {
+  title: 'jbjkka2 kbviu buibi',
+  body: 'ibin',
+  description: 'kvilbulibvi',
+};
+
+// tem test article
+const updateArticle = {
+  title: 'jbjkka2 kbviu buibi updated',
+  body: 'ibin update1 cc',
+  description: 'updfated kvilbulibvi',
+};
+
+let testSlug = '';
+let token = '';
+
+describe('Articles controller', () => {
+  describe('createArticle()', () => {
+    // get token to use for article route testing
+    it('', (done) => {
+      chai.request(server).post('/api/users').set('Accept', 'application/json').send(user)
+        .end((err, resp) => {
+          const userToken = resp.body.user.token;
+          token = userToken;
+          done();
+        });
+    });
+
+    it('should create and return the created article object', (done) => {
+      chai.request(server)
+        .post('/api/articles')
+        .set('Accept', 'application/json')
+        .set('Authorization', `Bearer ${token}`)
+        .set('Content-Type', 'application/json')
+        .send(Article)
+        .end((err, res) => {
+          res.should.have.status(201);
+          res.body.should.be.a('object');
```

Comment:
In lines `57 - 66` you can use destructuring to extract `article` from `res.body` so you don't have to repeat that.

Same thing with lines `119-126` and `156 - 163`

Reply:
Done.
Thanks.